### ISSUE TITLE
Can't link C targets with multiple archives

### ISSUE THREAD

**Issue by cstrahan** [bazelbuild/rules_go#1820]
It is currently not possible to link with targets possessing more than a single archive of the same type (i.e. file extension):

https://github.com/bazelbuild/rules_go/blob/109c520465fcb418f2c4be967f3744d959ad66d3/go/private/rules/cgo.bzl#L103-L116

My use case involves linking with [Envoy](https://github.com/envoyproxy/envoy), which has a dependency on `libevent`. The Envoy build system builds `event.a` and `event_pthreads.a` within the same target; only the former gets linked, resulting in the following missing symbol error:

```
Linking of rule '//bindings:go_default_library._cgo_.o' failed (Exit 1): envoy_cc_wrapper failed: error executing command 
  (cd /home/cstrahan/.cache/bazel/_bazel_cstrahan/0e3896d6d23d3761a1e0a1ef1f9e1eac/execroot/gm_proxy && \
  exec env - \
    PATH=/usr/local/bin:/home/cstrahan/.nix-profile/bin:/home/cstrahan/.nix-profile/lib/kde4/libexec:/nix/var/nix/profiles/default/bin:/nix/var/nix/profiles/default:/nix/var/nix/profiles/default/lib/kde4/libexec:/usr/local/sbin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/cstrahan/go/bin:/usr/local/go/bin \
    PWD=/proc/self/cwd \
  /home/cstrahan/.cache/bazel/_bazel_cstrahan/0e3896d6d23d3761a1e0a1ef1f9e1eac/external/local_config_cc/extra_tools/envoy_cc_wrapper -o bazel-out/k8-dbg/bin/bindings/go_default_library._cgo_.o -static-libstdc++ -static-libgcc -static-libstdc++ -static-libgcc -pthread -static-libstdc++ -static-libgcc -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread -pthread '-fuse-ld=gold' -Wl,-no-as-needed -Wl,-z,relro,-z,now -B/usr/bin -pass-exit-codes -Wl,@bazel-out/k8-dbg/bin/bindings/go_default_library._cgo_.o-2.params)

Use --sandbox_debug to see verbose messages from the sandbox
external/envoy/source/common/event/libevent.cc:16: error: undefined reference to 'evthread_use_pthreads'
collect2: error: ld returned 1 exit status
Target //:gm-proxy failed to build
INFO: Elapsed time: 7.850s, Critical Path: 7.62s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

---

Envoy itself does not have any issue building; could this be classified as a bug/limitation in `rules_go`, and therefore something that could be fixed?

**Comment by cstrahan** [comment]
It looks like there's some other (possibly somewhat related) issue at hand. I'm closing for now while I try to work out what exactly is going on.

**Comment by cstrahan** [comment]
For posterity, the problem was due to the fact that the Envoy build system builds `libevent` (and several other libs) in such a way that Bazel can't properly handle cache invalidation. So I update Envoy, which included a new version of `libevent`, but the build kept using the old version. The old version of `libevent` has `pthreads` support split off in a separate archive, and so the build failed due to the missing symbol. All I had to do was run `bazel clean --expunge` to fix the build.
